### ISSUE TITLE
Inventory subtrees are treated as separate "hardware" items

### ISSUE THREAD

**Issue by AlexanderAmelkin** [openbmc/phosphor-webui#6]
Some nodes under `/xyz/openbmc_project/inventory/system/chassis/` have subtrees.
For example, there are:
  * `/xyz/openbmc_project/inventory/system/chassis/bmc` and
  * `/xyz/openbmc_project/inventory/system/chassis/bmc/activation`

The latter one is a part of `bmc` information (a link to active software definition), but in WebUI it is displayed as a separate component under `Hardware` section:

![webui-hardware-activation](https://user-images.githubusercontent.com/4518124/41610768-b8b3dc26-73f6-11e8-9159-16d4570a5a16.png)

I believe, such entries must be put under their respective parent nodes somehow.

**Comment by AlexanderAmelkin** [comment]
I believe the problem here is much deeper than just phosphor-webui. The root cause is that a naturally tree-like structure is implemented in a naturally tree-friendly notation (JSON) as a plain list.
I believe that if phosphor-rest-server (or even phosphor-inventory-manager) presented the inventory as a tree, not as a plain list.

**Comment by iftekharuli** [comment]
I agree @AlexanderAmelkin. The time I developed it, we wanted to simplify the data with a tree like hierarchy order to future poof any info that might come in different environments. But maybe it is time to rethink and re-test this solution to make it more useful to users. The design team is taking it into consideration :)

**Comment by AlexanderAmelkin** [comment]
@ryanarnell, any progress on this?

**Comment by AlexanderAmelkin** [comment]
@gtmills, anybody?

**Comment by gtmills** [comment]
@jandraa @derick-montague @AlexanderAmelkin Is there a design that better handles parent /children inventory items?
